operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/Cleaner.java

```diff
@@ -0,0 +1,15 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+/**
+ * DependentResource can implement this interface to denote it requires explicit logic to clean up
+ * resources.
+ *
+ * @param <P> primary resource type
+ */
+@FunctionalInterface
+public interface Cleaner<P extends HasMetadata> {
+  void cleanup(P primary, Context<P> context);
```

Comment:
How that would work with more Dependent Resources returning different Delete Controls?

Reply:
Good point. Ultimately, though, I think it will make sense for dependent resources to return more information so that the main reconciler can make decisions based on the returned status…

Reply:
Yes, for now only the state is available, at that time. Although we changed the `reconcile() `  to return `ReconcileResult` this result is not available. Not sure if I would push this further ATM, but would be make those results also available in the future.